src/sources/journald.rs

```diff
@@ -95,55 +94,82 @@ type Matches = HashMap<String, HashSet<String>>;
 
 /// Configuration for the `journald` source.
 #[configurable_component(source("journald"))]
-#[derive(Clone, Debug, Default)]
-#[serde(deny_unknown_fields, default)]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
 pub struct JournaldConfig {
```

Comment:
It looks to me like all of the configuration settings are indeed optional, so I think you can just keep the serde default derivation at the struct level, and remove it from the fields that don't have a specific default to apply.

Reply:
@tobz what's your thought here, most of the fields use the default value (I think two bools aren't)

Reply:
So generally, the thought is that it's clearer to define per-field defaults -- even though it's more verbose than just deriving from the overall `Default` output of the struct -- because it's closer to the definition/description of the field.

Additionally, some `Default` impls take liberties to generate configurations that are usable for testing purposes and so on, and we really don't want to have those mistakenly flow into the resulting configuration schema.

Practically, as well, we use the "define defaults at the per-field level" approach in a majority of components, so we should make `journald` match that approach for consistency's sake.